### PR TITLE
[new module] Azure CDN Profile module

### PR DESCRIPTION
**Pull request by caohai**
- add azure_rm_cdnprofile module
- add azure_rm_cdnprofile_facts module
- add integration test for these two modules
- add cdn management client in azure_rm_common
- add necessary pkg info

### CODE REVIEW

File: library/azure_rm_cdnprofile.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+        default: present
+        choices:
+            - absent
+            - present
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+    sku:
+        description:
+            - The pricing tier (defines a CDN provider, feature list and rate) of the CDN profile.
```
Comment (yungezz) [inline library/azure_rm_cdnprofile.py]: better to add  a link to support values

File: library/azure_rm_cdnprofile.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+        name: cdntest
+        state: absent
+'''
+RETURN = '''
+state:
+    description: Current state of the CDN profile
+    returned: always
+    type: dict
+    example:
+            id: /subscriptions/685ba005-af8d-4b04-8f16-a7bf38b2eb5a/resourcegroups/cdntest/providers/Microsoft.Cdn/profiles/cdntest
```
Comment (yungezz) [inline library/azure_rm_cdnprofile.py]: remove your subs id

File: library/azure_rm_cdnprofile.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+            sku=Sku(name=self.sku),
+            tags=self.tags
+        )
+
+        try:
+            poller = self.cdn_management_client.profiles.create(self.resource_group, self.name, parameters)
+            response = self.get_poller_result(poller)
+            return cdnprofile_to_dict(response)
+        except ErrorResponseException as exc:
+            self.log('Error attempting to create Azure CDN profile instance.')
```
Comment (yungezz) [inline library/azure_rm_cdnprofile.py]: duplicate log

File: library/azure_rm_cdnprofile.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+            else:
+                self.log('Results : {0}'.format(response))
+                update_tags, response['tags'] = self.update_tags(response['tags'])
+
+                if response['provisioning_state'] == "Succeeded":
+                    if update_tags:
+                        to_be_updated = True
+
+                if to_be_updated:
```
Comment (yungezz) [inline library/azure_rm_cdnprofile.py]: no need to compare endpoint id list etc?

Comment (caohai) [reply library/azure_rm_cdnprofile.py]: CDN Profile itself doesn't have a list of endpoints.

**Pull request merged**